### PR TITLE
fix(url-sync): make URLSync consistent even if search is tampered

### LINKED ISSUE #2523: urlSync combined with specific searchFunction does not updates on first letter
**Do you want to request a *feature* or report a *bug*?**

bug

Test case: https://jsfiddle.net/8bqdcrg3/8/, this console.log the location.href continuously.

**Bug: What is the current behavior?**

When using this searchFunction:

```
{
  searchFunction: function(helper) {
    var query = helper.state.query;
    // with this code the url is not updated on first letter
    // while the search is being done
    if (query === '') {
	    return;
    }
    helper.search();
  },
}
```

The urlSync will not be triggered for the first typed letter, but only at the second one.

**Bug: What is the expected behavior?**

urlSync must be triggered for the first letter

**Bug: What browsers are impacted? Which versions?**

all

**Bug: What is the proposed solution?**

I don't know

**What is the version you are using? Always use the latest one before opening a bug issue.**

2/

### PR DESCRIPTION
**Pull request by bobylito**
When there is a search function that prevents the initial rendering,
the first actual change should be able to update the URL.

Fix #2523

### CODE REVIEW

**Comment by bobylito** [comment]
The dev-novel should be updated in order to be able to see the action logger :)

File: src/lib/url-sync.js [hunk 1]
```diff
@@ -132,11 +136,22 @@ class URLSync {
...
+  render({ helper, state }) {
     if (this.firstRender) {
       this.firstRender = false;
       this.onHistoryChange(this.onPopState.bind(this, helper));
-      helper.on('change', state => this.renderURLFromState(state));
+      helper.on('change', s => this.renderURLFromState(s));
+
+      const initStateQs = this.getQueryString(this.initState);
+      const stateQs = this.getQueryString(state);
+      console.log(initStateQs, stateQs);
```
Comment (samouss) [inline src/lib/url-sync.js]: Clean the `console.log` call

**Comment by bobylito** [comment]
I added a command to run eslint with prettier because I don't like autoreformat and it adds a dependency in the editor.

**Comment by bobylito** [comment]
I updated the code @samouss :)

**Comment by bobylito** [comment]
Thanks for the review @samouss I'm just waiting for the next version of dev-novel

**Pull request merged**